### PR TITLE
core: Add possibility to distinguish between auth cancelled and no creds

### PR DESCRIPTION
**Pull request by mfleisz**
Currently if the authentication callback returns `FALSE` the utils function handle this as scenario as no credentials provided (returns `AUTH_NO_CREDENTIALS)`.

This PR introduces a new `auth_status` called `AUTH_CANCELLED` that is returned if the authentication callback returns `FALSE`. If the callback returns `TRUE` and username or password are empty the util function will continue to return `AUTH_NO_CREDENTIALS`.

The PR also fixes some incorrect returns in RPC over HTTP gateway code.

### CODE REVIEW

**Comment by mfleisz** [comment]
@freerdp-bot test

File: libfreerdp/core/gateway/rpc_bind.c [hunk 1]
```diff
@@ -136,6 +136,9 @@ static int rpc_bind_setup(rdpRpc* rpc)
 		case AUTH_SUCCESS:
 		case AUTH_SKIP:
 			break;
+		case AUTH_CANCELLED:
+			freerdp_set_last_error_log(instance->context, FREERDP_ERROR_CONNECT_CANCELLED);
+			return FALSE;
```
Comment (akallabeth) [inline libfreerdp/core/gateway/rpc_bind.c:141]: wrong return type, function returns `int`

File: libfreerdp/core/gateway/ncacn_http.c [hunk 2]
```diff
@@ -161,21 +161,24 @@ BOOL rpc_ncacn_http_auth_init(rdpContext* context, RpcChannel* channel)
...
 	if (!credssp_auth_init(auth, AUTH_PKG, tls->Bindings))
-		return TRUE;
+		return FALSE;
 
 	if (sspi_SetAuthIdentityA(&identity, settings->GatewayUsername, settings->GatewayDomain,
 	                          settings->GatewayPassword) < 0)
-		return TRUE;
+		return FALSE;
 
 	credssp_auth_setup_client(auth, "HTTP", settings->GatewayHostname, &identity, NULL);
```
Comment (akallabeth) [inline libfreerdp/core/gateway/ncacn_http.c:183]: not your code, but missing return check

**Review by akallabeth — CHANGES_REQUESTED** [review]
Overall LGTM except the wrong return value

**Review by akallabeth — APPROVED** [review]

File: libfreerdp/core/gateway/ncacn_http.c (hunk 2 above)
Comment (mfleisz) [reply libfreerdp/core/gateway/ncacn_http.c]: @akallabeth thx for fixing 👍

**Pull request merged**